persistence/sql/persister_oauth2.go

```diff
@@ -252,10 +253,20 @@ func (p *Persister) deleteSessionBySignature(ctx context.Context, signature stri
 	signature = p.hashSignature(signature, table)
 
 	/* #nosec G201 table is static */
-	return sqlcon.HandleError(
-		p.Connection(ctx).
-			RawQuery(fmt.Sprintf("DELETE FROM %s WHERE signature=?", OAuth2RequestSQL{Table: table}.TableName()), signature).
-			Exec())
+	if err := p.Connection(ctx).
+		RawQuery(fmt.Sprintf("DELETE FROM %s WHERE signature=?", OAuth2RequestSQL{Table: table}.TableName()), signature).
+		Exec(); errors.Is(err, sql.ErrNoRows) {
+		return errorsx.WithStack(fosite.ErrNotFound)
+	} else if err := sqlcon.HandleError(err); err != nil {
+		if errors.Is(err, sqlcon.ErrConcurrentUpdate) {
+			return errors.Wrap(fosite.ErrSerializationFailure, err.Error())
+		} else if strings.Contains(err.Error(), "Error 1213") { // InnoDB Deadlock?
```

Comment:
Could we handle this in `sqlcon.HandleError` instead :)

Reply:
fixed at [1ad6caa](https://github.com/ory/hydra/pull/3207/commits/1ad6caa29fdf35cff6a5ef0f838d7aa3984e30e4)!